trio/_core/_entry_queue.py

```diff
@@ -63,14 +72,16 @@ async def kill_everything(exc):
                     # system nursery is already closed.
                     # TODO(2020-06): this is a gross hack and should
                     # be fixed soon when we address #1607.
-                    _core.current_task().parent_nursery.start_soon(kill_everything, exc)
+                    parent_nursery = _core.current_task().parent_nursery
+                    assert parent_nursery is not None
```

Comment:
Instead of an assert, it might be better to manually raise an `AssertionError`, so we can chain with `exc` not the `RuntimeError`. Shouldn't actually happen though.